internal/command/console/console.go

```diff
@@ -393,6 +393,15 @@ func determineEphemeralConsoleMachineGuest(ctx context.Context) (*api.MachineGue
 			return nil, fmt.Errorf("invalid CPU kind '%s'; this is a bug", desiredGuest.CPUKind)
 		}
 
+		// use vm.memory_mb from fly.toml (if present) as a lower bound
+		appConfig.Flatten("console")
```

Comment:
hmm, this is a NOOP line. `AppConfig.Flatten()` returns a copy of appConfig flatted for the process group.

honestly, much of the code in this function could use a simplification by reusing `internal.flag.GetMachineGuest()` and `internal.appconfig.Config.toMachineGuest()` functions.

Likely as part of `ToConsoleMachineConfig()` function from `internal/appconfig/machines.go`  file